tests/soter/test_vectors/soter_kdf.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2020 Cossack Labs Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"fmt"
+	"strings"
+)
+
+var key = []byte{0xAD, 0xEA, 0xDB, 0xEE, 0x01, 0x02, 0x03, 0x04}
+var label = "ZRTP KDF test vectors"
+var context = []byte{0x43, 0x83, 0xF3, 0x27, 0x0E, 0x11}
+var lengths = []int{1, 2, 4, 8, 16, 32}
+
+// A la RFC 6189 with HMAC-SHA-256
+// KDF(KI, label, context, L) = HMAC(KI, i || label || 0x00 || context || L)
+func zrtpKDF(key []byte, label string, context []byte, length int) []byte {
+	if !(1 <= length && length <= 32) {
```

Comment:
why ony [2, 31] range?

Reply:
Soter does not allow zero output length and it does not make much sense, so it's a safeguard.

32 is the maximum allowed length of KDF result — the output length of underlying HMAC-SHA-256 function. This KDF does not output a stream of key bytes, it’s limited to a single hash function output.

Reply:
oh, I missed "!" after "if". thanks, understood